src/main/java/com/labsynch/labseer/service/LsThingServiceImpl.java

```diff
@@ -2474,50 +2474,56 @@ else if (query.getRecordedDateLessThan() != null){
+						secondItxPredicates.add(secondThingLabelKind);
+					}
+					if (interaction.getThingLabelText() != null){
+						if (interaction.getOperator() != null){
+							if (interaction.getOperator().equals("=")){
+								Predicate secondThingLabelEquals = criteriaBuilder.equal(secondThingLabel.<String>get("labelText"), interaction.getThingLabelText());
+								secondItxPredicates.add(secondThingLabelEquals);
+							}else if (interaction.getOperator().equals("!=")){
+								Predicate secondThingLabelNotEquals = criteriaBuilder.notEqual(secondThingLabel.<String>get("labelText"), interaction.getThingLabelText());
+								secondItxPredicates.add(secondThingLabelNotEquals);
+							}else if(interaction.getOperator().equals("~")){
+								Predicate secondThingLabelLike = criteriaBuilder.like(criteriaBuilder.lower(secondThingLabel.<String>get("labelText")), '%' + interaction.getThingLabelText().toLowerCase() + '%');
+								secondItxPredicates.add(secondThingLabelLike);
+							}else if(interaction.getOperator().equalsIgnoreCase("like")){
+								Predicate secondThingLabelLike = criteriaBuilder.like(criteriaBuilder.lower(secondThingLabel.<String>get("labelText")), '%' + interaction.getThingLabelText().toLowerCase() + '%');
+								secondItxPredicates.add(secondThingLabelLike);
+							}else if(interaction.getOperator().equals("!~")){
+								Predicate secondThingLabelNotLike = criteriaBuilder.notLike(criteriaBuilder.lower(secondThingLabel.<String>get("labelText")), '%' + interaction.getThingLabelText().toLowerCase() + '%');
+								secondItxPredicates.add(secondThingLabelNotLike);
+							}else if(interaction.getOperator().equals(">")){
+								Predicate secondThingLabelGreaterThan = criteriaBuilder.greaterThan(secondThingLabel.<String>get("labelText"), interaction.getThingLabelText());
+								secondItxPredicates.add(secondThingLabelGreaterThan);
+							}else if(interaction.getOperator().equals(">=")){
+								Predicate secondThingLabelGreaterThan = criteriaBuilder.greaterThanOrEqualTo(secondThingLabel.<String>get("labelText"), interaction.getThingLabelText());
+								secondItxPredicates.add(secondThingLabelGreaterThan);
+							}else if(interaction.getOperator().equals("<")){
+								Predicate secondThingLabelLessThan = criteriaBuilder.lessThan(secondThingLabel.<String>get("labelText"), interaction.getThingLabelText());
+								secondItxPredicates.add(secondThingLabelLessThan);
+							}else if(interaction.getOperator().equals("<=")){
+								Predicate secondThingLabelLessThan = criteriaBuilder.lessThanOrEqualTo(secondThingLabel.<String>get("labelText"), interaction.getThingLabelText());
+								secondItxPredicates.add(secondThingLabelLessThan);
+							}else{
+								Predicate secondThingLabelLike = criteriaBuilder.like(secondThingLabel.<String>get("labelText"), '%' + interaction.getThingLabelText() + '%');
```

Comment:
Same thing here. Should make this case insensitive IMO.

Reply:
Interesting that its the opposite here. fixing to be the same (both lower)